include/sys/gmon.h

```diff
@@ -0,0 +1,160 @@
+#define _SYS_GMON_H_
+
+void init_kgprof(void);
+
+/*
+ * Structure prepended to gmon.out profiling data file.
+ */
+typedef struct gmonhdr {
+  u_long lpc;   /* base pc address of sample buffer */
+  u_long hpc;   /* max pc address of sampled buffer */
+  int ncnt;     /* size of sample buffer (plus this header) */
+  int version;  /* version number */
+  int profrate; /* profiling clock rate */
+  int spare[3]; /* reserved */
+} gmonhdr_t;
+extern gmonhdr_t _gmonhdr;
+
+#define GMONVERSION 0x00051879
+
+/*
+ * histogram counters are unsigned shorts (according to the kernel).
+ */
+typedef unsigned short HISTCOUNTER;
+
+/*
+ * fraction of text space to allocate for histogram counters here, 1/2
+ */
+#ifndef HISTFRACTION
+#define HISTFRACTION 2
+#endif /* HISTFRACTION */
+
+#define INSTR_GRANULARITY HISTFRACTION * sizeof(HISTCOUNTER)
+
+/*
+ * Fraction of text space to allocate for from hash buckets.
+ * The value of HASHFRACTION is based on the minimum number of bytes
+ * of separation between two subroutine call points in the object code.
+ * Given MIN_SUBR_SEPARATION bytes of separation the value of
+ * HASHFRACTION is calculated as:
+ *
+ *	HASHFRACTION = MIN_SUBR_SEPARATION / (2 * sizeof(short) - 1);
+ *
+ * For example, on the VAX, the shortest two call sequence is:
+ *
+ *	calls	$0,(r0)
+ *	calls	$0,(r0)
+ *
+ * which is separated by only three bytes, thus HASHFRACTION is
+ * calculated as:
+ *
+ *	HASHFRACTION = 3 / (2 * 2 - 1) = 1
+ *
+ * Note that the division above rounds down, thus if MIN_SUBR_FRACTION
+ * is less than three, this algorithm will not work!
+ *
+ * In practice, however, call instructions are rarely at a minimal
+ * distance.  Hence, we will define HASHFRACTION to be 2 across all
+ * architectures.  This saves a reasonable amount of space for
+ * profiling data structures without (in practice) sacrificing
+ * any granularity.
+ */
+#define HASHFRACTION 2
+
+/*
+ * percent of text space to allocate for tostructs with a minimum.
+ */
+#define ARCDENSITY 2
+#define MINARCS 50
+#define MAXARCS ((1 << (unsigned int)(8 * sizeof(HISTCOUNTER))) - 2)
+
+typedef struct tostruct {
+  u_long selfpc;
+  long count;
+  u_short link;
+  u_short pad;
+} tostruct_t;
+
+/*
+ * a raw arc, with pointers to the calling site and
+ * the called site and a count.
+ */
+struct rawarc {
+  u_long raw_frompc;
+  u_long raw_selfpc;
+  long raw_count;
+};
+
+/*
+ * The profiling data structures are housed in this structure.
+ */
+typedef struct gmonparam {
+  int state;
+  HISTCOUNTER *kcount;
+  u_long kcountsize;
+  u_short *froms;
+  u_long fromssize;
+  struct tostruct *tos;
+  u_long tossize;
+  long tolimit;
+  u_long lowpc;
+  u_long highpc;
+  u_long textsize;
+  u_long hashfraction;
+} gmonparam_t;
+extern gmonparam_t _gmonparam;
+
+/*
+ * Possible states of profiling.
+ */
+typedef enum {
+  GMON_PROF_ON = 0,
+  GMON_PROF_BUSY = 1,
+  GMON_PROF_ERROR = 2,
+  GMON_PROF_OFF = 3,
+  GMON_PROF_NOT_INIT = 4,
+} gmon_flags_t;
```

Comment:
These aren't flags (i.e. they're not meant to be bitwise-ORed), so `gmon_state_t` is a better name IMO.